docs/src/configuration.md

```diff
@@ -52,64 +48,115 @@ with:
 $ mpiexecjl --project=/path/to/project -n 20 julia script.jl
 ```
 
-## Using a system-provided MPI
+## Using MPIPreferences.jl
+
+MPI.jl uses [Preferences.jl](https://github.com/JuliaPackaging/Preferences.jl) to
+allow the user to choose which MPI implementation to use for a project. This provides
+a single source of truth that can be used for JLL packages (Julia packages providing C libraries)
+that link against MPI, localizes the choice of MPI implementation to a project.
+
+Users can use the provided [`use_system_binary`](@ref) or [`use_jll_binary`](@ref)
+to switch MPI implementations. By default, the JLL-provided binaries are used.
+
+### Migration from MPI.jl `v0.19`
+
+Prior to MPI.jl `v0.20` environment variables were used to configure which MPI
+library to use. These have now been removed and have no effect anymore:
 
-### Requirements
+- `JULIA_MPI_BINARY`
+- `JULIA_MPIEXEC`
+- `JULIA_MPI_INCLUDE_PATH`
+- `JULIA_MPI_CFLAGS`
+- `JULIA_MPICC`
+
+### Using a system-provided MPI backend
+
+#### Requirements
 
 MPI.jl requires a shared library installation of a C MPI library, supporting the MPI 3.0
 standard or later.
 
-### Building
+### Configuration
 
-To use the the system MPI, set the environment variable `JULIA_MPI_BINARY=system` and run
-`Pkg.build("MPI")`. This can be done by:
-```
-julia --project -e 'ENV["JULIA_MPI_BINARY"]="system"; using Pkg; Pkg.build("MPI"; verbose=true)'
+To use the system MPI library, run `MPI.use_system_binary()`.
```

Comment:
```julia
julia> MPI.use_system_binary()
ERROR: UndefVarError: use_system_binary not defined
Stacktrace:
 [1] getproperty(x::Module, f::Symbol)
   @ Base ./Base.jl:35
 [2] top-level scope
   @ REPL[4]:1
```
I guess you either want to import the function from `MPIPreferences` (and maybe do `const use_system_binary = MPIPreferences.use_system_binary` to enable REPL completion), or mention here `MPIPreferences.use_system_binary`

Reply:
☝️ good point.

Reply:
Uh, I missed in this PR you're importing the names from MPIPreferences, but I think it'd still be good to make REPL completion work, it's a small quality-of-life improvement